### ISSUE TITLE
Different results for dl4j versions 0.8.0 and 0.9.1

### ISSUE THREAD

**Issue by MariyaF** [deeplearning4j/deeplearning4j#5037]
Issue Description

I switched from the library version 0.8.0 to the version 0.9.1. Parameters of ANN did not change. Retrained with new version. My RMSE change from 0.1 to 0.6.
Why the results are so different for different versions?

Version Information

Deeplearning4j version from 0.8.0 to 0.9.1
platform information (OS, etc) windows 10
Contributing

Topology for 0.8.0

`
private MultiLayerConfiguration getDeepDenseLayerNetworkConfiguration() {
final int numHiddenNodes = 100;

    return new NeuralNetConfiguration.Builder()
            .seed(SEED)
            .iterations(ITERATIONS)
            .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
            .learningRate(LEARNING_RATE)
            .weightInit(WeightInit.XAVIER)
            .updater(Updater.NESTEROVS)
            .momentum(0.9)
            .list()
            .layer(0, new GravesLSTM.Builder().nIn(NUM_INPUTS).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(1, new GravesLSTM.Builder().nIn(numHiddenNodes).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(2, new RnnOutputLayer.Builder(LossFunctions.LossFunction.MSE).activation(Activation.SIGMOID).nIn(numHiddenNodes).nOut(NUM_OUTPUTS).build())
            .pretrain(false)
            .backprop(true)
            .build();
}`
Topology for 0.9.1

`
private MultiLayerConfiguration getDeepDenseLayerNetworkConfiguration() {
final int numHiddenNodes = 100;

    return new NeuralNetConfiguration.Builder()
            .seed(SEED)
            .iterations(ITERATIONS)
            .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
            .learningRate(LEARNING_RATE)
            .weightInit(WeightInit.XAVIER)
            .updater(new Nesterovs(0.9))
            .list()
            .layer(0, new GravesLSTM.Builder().nIn(NUM_INPUTS).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(1, new GravesLSTM.Builder().nIn(numHiddenNodes).nOut(numHiddenNodes).activation(Activation.HARDTANH).build())
            .layer(2, new RnnOutputLayer.Builder(LossFunctions.LossFunction.MSE).activation(Activation.SIGMOID).nIn(numHiddenNodes).nOut(NUM_OUTPUTS).build())
            .pretrain(false)
            .backprop(true)
            .build();
}`

**Comment by agibsonccc** [comment]
Please try the latest version and tell us what the expected result is.

**Comment by MariyaF** [comment]
I tried the last version already (1.0.0-alpha).
RMSE like in version 0.9.1.

I expected that with the version change the model will not change and when I retrain I will get the same RMSE.

Which changes when changing versions.

**Comment by treo** [comment]
It depends on how exactly you've configured your model. If you've left a lot of things on default, you may be seeing the effect of defaults changing.

**Comment by AlexDBlack** [comment]
Furthermore we can't rule out a bug in 0.8.0 that was causing this...

That said, in 0.9.1 there was an occasional issue when using both the ```.learningRate(double)``` and the ```.updater(IUpdater)``` configuration methods. I'd recommend trying again setting the LR on the Nesterovs object too.

Note also that a number of the RNN examples (with very similar configs to yours) have existed since long before 0.8.0 and we check them every release, and there haven't been any changes/issues there.

At any rate, there's no way we can reproduce this or debug it without (a) the data you are using, and (b) information on exactly how you are calculating the RMSE that changes between versions.
